diff --git a/packages/ui/Dockerfile b/packages/ui/Dockerfile
index 43e0d2c..3cc33c3 100644
--- a/packages/ui/Dockerfile
+++ b/packages/ui/Dockerfile
@@ -7,15 +7,16 @@ COPY deploy ./deploy
 COPY deploy/nginx /etc/nginx
 COPY deploy/supervisor /etc/supervisor
 
-COPY ./ytsaurus-ui-*.tgz ./
+COPY . .
 
 ARG UI_VERSION
 ARG DEV
 
-RUN UI_PACKAGE=$(test -n "${DEV}" && echo ./ytsaurus-ui-${UI_VERSION}.tgz || echo @ytsaurus/ui@${UI_VERSION}) \
-        && echo "\n  UI_VERSION=${UI_VERSION} \n  DEV=${DEV} \n  UI_PACKAGE=${UI_PACKAGE} \n" >&2 \
-        && npm install --legacy-peer-deps $UI_PACKAGE \
-        && mv node_modules/@ytsaurus/ui . && mv ui/* .
+# build app
+RUN npm ci -q --no-progress --include=dev --also=dev
+RUN npm run build
+RUN npm prune --production
+RUN rm -rf assets src /tmp/* /root/.npm
 
 EXPOSE 80
 
diff --git a/packages/ui/deploy/scripts/preflight.sh b/packages/ui/deploy/scripts/preflight.sh
index b3e7c1f..ebc0a76 100644
--- a/packages/ui/deploy/scripts/preflight.sh
+++ b/packages/ui/deploy/scripts/preflight.sh
@@ -7,7 +7,6 @@ sed -e "s/%APP_HTTP_PORT%/${APP_HTTP_PORT:-80}/" -i /etc/nginx/sites-enabled/*.c
 
 mkdir -p /opt/app/secrets /opt/app/dist/run
 
-chown -R app /opt/app/secrets
 chown app /opt/app/dist/run
 
 supervisorctl start node
