diff --git a/packages/ui/src/server/components/yt-auth.ts b/packages/ui/src/server/components/yt-auth.ts
--- a/packages/ui/src/server/components/yt-auth.ts
+++ b/packages/ui/src/server/components/yt-auth.ts
@@ -9,12 +9,13 @@ export function isYtAuthEnabled(config: AppConfig) {
 
 export function YTAuthLogout(res: Response) {
     const clusters = getClustersFromConfig();
+    const host = res.req.hostname;
 
     res.setHeader(
         'set-cookie',
-        [`${YT_CYPRESS_COOKIE_NAME}=deleted; Path=/; Max-Age=0;`].concat(
+        [`${YT_CYPRESS_COOKIE_NAME}=deleted; Path=/; Max-Age=0; Domain=${host}`].concat(
             Object.keys(clusters).map(
-                (cluster) => `${cluster}:${YT_CYPRESS_COOKIE_NAME}=deleted; Path=/; Max-Age=0;`,
+                (cluster) => `${cluster}:${YT_CYPRESS_COOKIE_NAME}=deleted; Path=/; Max-Age=0; Domain=${host};`,
             ),
         ),
     );
diff --git a/packages/ui/src/server/controllers/login.ts b/packages/ui/src/server/controllers/login.ts
--- a/packages/ui/src/server/controllers/login.ts
+++ b/packages/ui/src/server/controllers/login.ts
@@ -23,6 +23,7 @@ export async function handleLogin(req: Request, res: Response) {
 
         const {proxyBaseUrl} = getYTApiClusterSetup(ytAuthCluster);
         const requestUrl = `${proxyBaseUrl}/login`;
+        const host = req.hostname;
 
         const basicAuth = Buffer.from(`${username}:${password}`).toString('base64');
 
@@ -44,6 +45,7 @@ export async function handleLogin(req: Request, res: Response) {
                         if (headers['set-cookie']) {
                             headers['set-cookie'] = headers['set-cookie'].reduce<string[]>(
                                 (ret, item) => {
+                                    item = item.concat(`; Domain=${host}`);
                                     ret.push(item);
 
                                     if (item.startsWith(YT_CYPRESS_COOKIE_NAME)) {
diff --git a/packages/ui/src/ui/store/actions/cluster-params.ts b/packages/ui/src/ui/store/actions/cluster-params.ts
--- a/packages/ui/src/ui/store/actions/cluster-params.ts
+++ b/packages/ui/src/ui/store/actions/cluster-params.ts
@@ -169,6 +169,7 @@ export function updateCluster(
         });
 
         const rumId = new RumWrapper(cluster, RumMeasureTypes.CLUSTER_PARAMS);
+        const host = window.location.hostname;
         return rumId
             .fetch<AxiosResponse<ClusterInfoData>>(
                 YTApiId.ui_clusterInfo,
@@ -217,6 +218,7 @@ export function updateCluster(
                         YT.parameters.login = login;
                         dispatch({type: GLOBAL_PARTIAL, data: {login}});
                         Cookies.set(getXsrfCookieName(cluster), csrf_token);
+                        Cookies.set('CSRF-TOKEN', csrf_token, { domain: host })
                         return onUpdateEnd()
                             .then(() => {
                                 dispatch({
